TECHMANIA/Assets/Scripts/DiscordController.cs

```diff
@@ -23,8 +24,10 @@ public class DiscordController
 
     public static void Start ()
     {
-        if (discord != null || !SupportedOnCurrentPlatform()) return;
-        discord = new Discord.Discord(802017593086836767, (UInt64)Discord.CreateFlags.Default);
+        if (discord != null || !SupportedOnCurrentPlatform() || !Options.instance.discordRichPresence) return;
+        try {
+            discord = new Discord.Discord(802017593086836767, (UInt64)Discord.CreateFlags.NoRequireDiscord);
+        } catch {}
```

Comment:
Do we know what exceptions `Discord.Discord` may throw? Can we catch and handle them specifically? Or at least catch `Exception` and print it in console?

Same with `Dispose()`.

Reply:
`Discord.Discord` will throw this error if Discord is not running.
```
ResultException: InternalError
Discord.Discord..ctor (System.Int64 clientId, System.UInt64 flags) (at Assets/Plugins/DiscordGameSDK/Core.cs:1017)
DiscordController.Start () (at Assets/Scripts/DiscordController.cs:28)
Startup.Start () (at Assets/Scripts/Components/Main Menu/Startup.cs:18)
```

Reply:
And looks like `Dispose()` does not throw any error message.
But `discord.RunCallbacks()` throws this error if Discord is not running, this happens if Discord is closed while playing the game.
I'm working on new commit to fix this.
```
ResultException: NotRunning
Discord.Discord.RunCallbacks () (at Assets/Plugins/DiscordGameSDK/Core.cs:1056)
DiscordController.RunCallbacks () (at Assets/Scripts/DiscordController.cs:34)
DiscordUpdater.Update () (at Assets/Scripts/Components/DiscordUpdater.cs:7)
```